File: go/hamming/hamming.go
//Package hamming is what's up
package hamming

import "errors"

//Distance - well the hamming distance at least
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("sequences must be the same length")
	}
	count := 0
	for i := 0; i < len(a); i++ {
		if string(a[i]) != string(b[i]) {
			count++
		}
	}
	return count, nil
}

File: ruby/hamming/hamming.rb
# frozen_string_literal: true

# class Hamming
class Hamming
  def self.compute(str1, str2)
    count = 0
    raise ArgumentError, 'Sequences must be the same length' unless str1.size == str2.size

    (0..(str1.size - 1)).each { |i| str1[i] != str2[i] && count += 1 }
    count
  end
end
